src/App.jsx

```diff
@@ -1,39 +1,101 @@
+import {
+	BrowserRouter as Router,
+	Route,
+	Switch,
+	Redirect,
+} from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { useFetch } from './mixins/use-fetch';
+import axios from 'axios';
 import Header from './components/Header/Header';
 import Courses from './components/Courses/Courses';
+import CourseInfo from './components/Courses/CourseInfo';
+import Login from './components/Login/Login';
+import Registration from './components/Registration/Registration';
 import CreateCourse from './components/CreateCourse/CreateCourse';
-import mockedCoursesList from './data/mockedCoursesList';
-import mockedAuthorsList from './data/mockedAuthorsList';
-import { useState } from 'react';
+import Error from './components/Error/Error';
+import PrivateRoute from './components/Router/PrivateRoute';
+import isTokenExist from './mixins/token';
 
 const App = () => {
-	const [areCoursesVisible, setCoursesVisibility] = useState(true);
-	const [courses, setCourses] = useState(mockedCoursesList);
-	const [authors, setAuthors] = useState(mockedAuthorsList);
+	const [isLoggedIn, setIsLoggedIn] = useState(isTokenExist);
+	const [userName, setUserName] = useState('');
+	const courses = useFetch('http://localhost:3000/courses/all').data;
+	const authors = useFetch('http://localhost:3000/authors/all').data;
 
-	const createCourse = ({ newCourse, allAuthors }) => {
-		const updatedCoursesList = [...courses, newCourse];
-		setCourses(updatedCoursesList);
-		setAuthors(allAuthors);
-		setCoursesVisibility(!areCoursesVisible);
+	const handleLogout = () => {
+		axios
+			.delete('http://localhost:3000/logout', {
+				headers: {
+					Authorization: localStorage.getItem('userToken'),
+				},
+			})
+			.then(() => {
+				localStorage.removeItem('userToken');
+				setIsLoggedIn(false);
+				setUserName('');
+				window.location.href = '/login';
+			});
 	};
 
+	const handleLogin = () => {
+		setIsLoggedIn(true);
+	};
+
+	const createCourse = () => {
+		window.location.href = '/courses';
+	};
+	useEffect(() => {
+		if (isTokenExist) {
+			setIsLoggedIn(true);
+			axios
+				.get('http://localhost:3000/users/me', {
+					headers: {
+						Authorization: localStorage.getItem('userToken'),
+					},
+				})
+				.then((response) => {
+					setUserName(response.data.result.name);
+				});
+		}
+	}, [isLoggedIn]);
+
 	return (
-		<div className='App'>
-			<Header />
-			{areCoursesVisible ? (
-				<Courses
-					courses={courses}
-					authors={authors}
-					createCourse={() => setCoursesVisibility(!areCoursesVisible)}
-				/>
-			) : (
-				<CreateCourse
-					authors={authors}
-					createCourse={createCourse}
-					cancel={() => setCoursesVisibility(!areCoursesVisible)}
-				/>
-			)}
-		</div>
+		<Router>
+			<Header
+				isLoggedIn={isLoggedIn}
+				user={userName}
+				handleLogout={handleLogout}
+			/>
+			<Switch>
+				<Route
+					path='/login'
+					component={() => <Login handleLogin={handleLogin} />}
```

Comment:
hm, I can't add double quotes  in <Router /> attributes, I get the error from ESLint and Prettier. I think it's okay since we have here not base html but react component